server/src/test/java/io/spine/examples/shareaware/server/investment/SharesSaleTest.java

```diff
@@ -0,0 +1,312 @@
+            context().receivesCommand(sellCommand);
+            BalanceRecharged expected = balanceRechargedBy(sellCommand);
+
+            context().assertEvents()
+                     .withType(BalanceRecharged.class)
+                     .message(1)
+                     .isEqualTo(expected);
+        }
+    }
+
+    @Nested
+    @DisplayName("interact with `Investment`")
+    class InvestmentBehaviour {
+
+        @Test
+        @DisplayName("by reducing the number of available shares")
+        void state() {
+            Investment investment = setUpInvestment(context());
+            SellShares command = sellShareFrom(investment);
+            context().receivesCommand(command);
+            Investment expected = investmentAfter(command, investment);
+
+            context().assertState(investment.getId(), expected);
+        }
+
+        @Test
+        @DisplayName("by emitting the `SharesReserved` event")
+        void sharesReserved() {
+            Investment investment = setUpInvestment(context());
+            SellShares command = sellShareFrom(investment);
+            context().receivesCommand(command);
+            SharesReserved expected = sharesReservedBy(command);
+
+            context().assertEvent(expected);
+        }
+
+        @Test
+        @DisplayName("by emitting the `SharesReservationCompleted` event")
+        void sharesReservationCompleted() {
+            Investment investment = setUpInvestment(context());
+            SellShares command = sellShareFrom(investment);
+            context().receivesCommand(command);
+            SharesReservationCompleted expected =
+                    sharesReservationCompletedBy(command, investment);
+
+            context().assertEvent(expected);
+        }
+
+        @Test
+        @DisplayName("by emitting the `SharesReservationCanceled event " +
+                "after error in the shares market")
+        void sharesReservationCanceled() {
+            Investment investment = setUpInvestment(context());
+            SellShares command = sellShareFrom(investment);
+            RejectingMarket.switchToRejectionMode();
+            context().receivesCommand(command);
+            SharesReservationCanceled expected =
+                    sharesReservationCanceledAfter(command);
+
+            context().assertEvent(expected);
+            RejectingMarket.switchToEventsMode();
+        }
+
+        @Test
+        @DisplayName("by emitting the `InsufficientShares` rejection")
+        void insufficientShares() {
+            Investment investment = setUpInvestment(context());
+            SellShares command = sellMoreSharesThanIn(investment);
+            context().receivesCommand(command);
+            InsufficientShares expected = insufficientSharesCausedBy(command);
+
+            context().assertEvent(expected);
+        }
+    }
+
+    @Nested
+    @DisplayName("be led by shares sale process")
+    class SaleProcess {
+
+        @Test
+        @DisplayName("with state")
```

Comment:
Not sure I understand this scenario: 
```
`SharesSale` should ... be led by shares sale process ... with state
```
What does this even mean?

Please review.